### PR TITLE
add otel exporter to the loki historian pipeline

### PR DESCRIPTION
**Pull request by ErezLevip**
added an otlp exporter to the loki pipeline.

to reduced the size of the change and to reduce potential future conflicts i changed 3 files of the original code and added 1 that contains the entire client implementation.

1. The Historian module is producing loki streams with all the attributes that were relevant both on each entry (inside the json) and on the stream it self (things like org for example) i added a new client called otelLokiClient that implements the same interface.
I didn't implement  the query and check functions since they are not required and we dont really query (right now vector) for any state at the moment.

2. Implemented the currently in use grafana metrics (bytes written and duration, the rest is implemented around the push mechanism).

3.  Send every attribute to vector, handling of what is saved will be in VRL

Added values changes on the helm side
https://github.com/groundcover-com/groundcover-private/pull/8419/files

### CODE REVIEW

File: pkg/services/ngalert/state/historian/otel_loki_client.go [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+func (p *otelLokiClient) RangeQuery(ctx context.Context, logQL string, start, end, limit int64) (QueryRes, error) {
+	return QueryRes{}, fmt.Errorf("unsupported operation")
+}
+
+func (p *otelLokiClient) Push(ctx context.Context, s []Stream) (err error) {
+	const (
+		timerFailureCode = "500"
+		exportMethodName = "OtelExport"
+	)
+	p.once.Do(func() {
```
Comment (udiz) [inline pkg/services/ngalert/state/historian/otel_loki_client.go:82]: for clarity, let's put this in another function (`initClientConn` for example)

File: pkg/services/ngalert/state/historian/otel_loki_client.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+	_, err = p.client.Export(ctx, plogotlp.NewExportRequestFromLogs(logs))
+	if err != nil {
+		return fmt.Errorf("failed to export logs: %w", err)
+	}
+	p.metrics.WriteDuration.After(ctx, exportMethodName, timerFailureCode, exportStart)
+	p.metrics.BytesWritten.Add(float64(size))
+	return nil
+}
+
+func (p *otelLokiClient) pushRequestToLogs(sreams []Stream, defaultTimestamp time.Time) (plog.Logs, int, error) {
```
Comment (udiz) [inline pkg/services/ngalert/state/historian/otel_loki_client.go]: rename `defaultTimestamp` to `defaultObservedTimestmap`

Comment (udiz) [reply pkg/services/ngalert/state/historian/otel_loki_client.go]: that's acutally not even "default", it's just the observedTimestamp

File: pkg/services/ngalert/state/historian/otel_loki_client.go [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+		timestamp = pcommon.NewTimestampFromTime(entry.T)
+	} else {
+		timestamp = observedTimestamp
+	}
+
+	lr.SetTimestamp(timestamp)
+	recordAttributes[timestampAttribute] = timestamp.AsTime().Format(time.RFC3339Nano)
+
+	for k, v := range streamAttributes {
+		recordAttributes[k] = v
```
Comment (udiz) [inline pkg/services/ngalert/state/historian/otel_loki_client.go]: why not putting directly into the `lr.Attributes()` pcommon.Map?

File: pkg/services/ngalert/state/historian/otel_loki_client.go [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+
+func newOtlpGrpcConn(cfg OtelConfig) (conn *grpc.ClientConn, err error) {
+	const (
+		apiKeyHeader                 = "apikey"
+		defaultConnectionDialTimeout = 10 * time.Second
+	)
+	creds := insecure.NewCredentials()
+	if cfg.EnableTLS {
+		config := &tls.Config{
+			InsecureSkipVerify: cfg.TLSSkipVerify,
```
Comment (udiz) [inline pkg/services/ngalert/state/historian/otel_loki_client.go:287]: right now we support forcing verify on monitors? we had that issue with alligator in a client recently

File: pkg/services/ngalert/state/historian/otel_loki_client.go (hunk 1 above)
Comment (ErezLevip) [reply pkg/services/ngalert/state/historian/otel_loki_client.go:82]: changed to getClient with singleton in it

**Review by udiz — APPROVED** [review]

**Review by avivzgroundcover — APPROVED** [review]

**Pull request merged**